src/tools/clang.lua

```diff
@@ -84,6 +84,23 @@
 		return gcc.getwarnings(cfg)
 	end
 
+--
+-- Returns C/C++ system version related build flags
+--
+
+	function clang.getsystemversionflags(cfg)
+		local flags = {}
+
+		if cfg.system == p.MACOSX or cfg.system == p.IOS then
+			local minVersion = p.project.systemversion(cfg)
+			if (type (minVersion) == "string") and (string.match(minVersion, "^%d+%.%d+") ~= nil) then
```

Comment:
The type will always be string or nil, so `minVersion ~= nil` is how we normally test this kind of thing. Additionally, we don't validate the version string in xcode [here](https://github.com/premake/premake-core/blob/2c42771f3f66b3efb9c64fdf9a0f69b02496c62f/modules/xcode/xcode_common.lua#L1206-L1209), doing validation here would be inconsistent.

Reply:
1. Okay for the "~= nil" test. 
2. So, shall I remove the validation or shall I add it to Xcode ? If validation is kept, it may be useful to add a generic macox system version validation function somewhere in the core.